solver/src/solver/matcha_solver.rs

```diff
@@ -122,7 +122,7 @@ impl SingleOrderSolving for MatchaSolver {
         });
         let spender = swap.allowance_target;
 
-        settlement.with_liquidity(&order, swap.sell_amount)?;
+        settlement.with_liquidity(&order, order.full_execution_amount())?;
```

Comment:
Thanks for doing the fix! Do you happen to have the link to an issue? Where can I see the instance? I do not yet fully understand the problem

The full execution amount can be smaller than the sell_amount for buy orders, as because of the slippage matcha is selling a little bit more, just to have sufficient funds, in case slippage was suffered. 

I am okay with reducing the buffer like that, but then we also need change the price here:
```
let mut settlement = Settlement::new(hashmap! {
            order.sell_token => swap.buy_amount,
            order.buy_token => order.full_execution_amount(),
        });
        ```

Reply:
The executed amount should not depend on the slippage and is independent of the price. For sell orders, it is how much of the sell amount to use, and for buy order it is how much of he buy amount to use. Since these are fill-or-kill orders, we should always execute the full amount, regardless of the trading price.

I went into mode detail [here](https://gnosisinc.slack.com/archives/C020UHS2UEL/p1625586555012800).

Reply:
Ah cooL!
I was not aware of how handy the `with_liquidity` function is and that it works differently depending on the order type.  Love it!